tests/_test_msui/test_msui.py

```diff
@@ -328,3 +336,181 @@ def test_flight_track_io(self, mockload, mocksave, mockq, mocki, mockw):
         assert self.window.listFlightTracks.count() == 2
         assert os.path.exists(self.save_ftml)
         os.remove(self.save_ftml)
+
+
+class Test_MSUIMainWindow:
+    def test_storing_and_restoring(self, qtbot, mswms_server):
+        """Test the full scenario: create flight track, open TopView, modify settings,
+        save on close, and restore settings on reopen."""
+
+        parsed_url = urllib.parse.urlparse(mswms_server)
+        scheme, host, port = parsed_url.scheme, parsed_url.hostname, parsed_url.port
+        server_url = f"{scheme}://{host}:{port}"
+        logging.debug(server_url)
+
+        window = msui_mw.MSUIMainWindow()
+        window.show()
+
+        # create 1st flighttrack
+        window.create_new_flight_track()
+        assert window.listFlightTracks.count() == 1
+        flight_track = window.active_flight_track
+        assert "new flight track (1)" in window.activated_flight_tracks
+
+        waypoint_model = flight_track
+        waypoint_model.insertRows(0, 2, waypoints=[
+            ft.Waypoint(lat=34.44, lon=56.67, location="point1"),
+            ft.Waypoint(lat=77.77, lon=98.67, location="point2")
+        ])
+
+        # Open top view for first flight track
+        window.create_view("topview", flight_track)
+        assert window.listViews.count() == 1
+        top_view1 = window.listViews.item(0).window
+        assert top_view1.view_type == "Top View"
+
+        top_view1.cbChangeMapSection.setCurrentText("00 global (cyl)")
+        wms_settings1 = {
+            "url": server_url,
+            "layer": "ecmwf_EUR_LL015.PLRelHum01",
+            "level": "200.0",
+            "styles": "",
+            "init_time": "2012-10-17T12:00:00Z",
+            "valid_time": "2012-10-17T12:00:00Z",
+        }
+        top_view1.restore_wms_settings(wms_settings1)
+        qtbot.waitUntil(
+            lambda: top_view1.wms_control.multilayers.cbWMS_URL.currentText().rstrip('/') == server_url,
+            timeout=500
+        )
+
+        # create 2nd flighttrack
+        window.create_new_flight_track()
```

Comment:
maybe It would be better to use a filename parameter and use a name.

you maybe need an information in the view_settings.json to compare with the data of the waypoint table. e.g. path, url, ..